src/Symfony/Component/Validator/Constraints/NotJsonValidator.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+/**
+ * @author Imad ZAIRIG <imadzairig@gmail.com>
+ */
+class NotJsonValidator extends ConstraintValidator
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function validate($value, Constraint $constraint)
+    {
+        if (!$constraint instanceof NotJson) {
+            throw new UnexpectedTypeException($constraint, __NAMESPACE__.'\NotJson');
+        }
+
+        if (!$this->isJson($value)) {
+            $this->context->buildViolation($constraint->message)
+                ->setParameter('{{ value }}', $this->formatValue($value))
+                ->setCode(NotJson::NOT_JSON_ERROR)
+                ->addViolation();
+        }
+    }
+
+    /**
+     * @param string $string
+     *
+     * @return bool
+     */
+    protected function isJson($string)
+    {
+        json_decode($string);
+
+        return JSON_ERROR_NONE == json_last_error();
```

Comment:
Can you make this a strict type check? I'd hate to see a bug because for some odd reason this would return false (most likely won't happen, but you can't be too sure with php)

Reply:
yes, I've added the strict comparison.